azure/services/scalesets/scalesets.go

```diff
@@ -791,6 +814,51 @@ func getSecurityProfile(vmssSpec azure.ScaleSetSpec, sku resourceskus.SKU) (*com
 	}, nil
 }
 
+func (s *Service) generateDiagnosticsProfile(vmssSpec azure.ScaleSetSpec) (*compute.DiagnosticsProfile, error) {
+	if vmssSpec.DiagnosticsProfile.Boot == nil {
```

Comment:
same comments as VM spec for this function

Reply:
given there is duplicate code here, could we instead share a common function that converts a `infrav1.DiagnosticsProfile`  into a `compute.DiagnosticsProfile` with the shared business logic? This new func could go in https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/main/azure/converters

Reply:
Nice idea, logic centralized to `converters`.